### DISCUSSION TITLE
Connection to node.socket

### DISCUSSION THREAD

**Discussion by djcyr** [bloxbean/cardano-client-lib#180] [Ideas]
Is it possible to have a BackendService connect to a local daedalus or cardano-node instance on the node.socket path?

**Comment by satran004** [comment]
@djcyr Right now there is no out of box BackendService to do that. 

Let me know what kind of info you want to fetch from the node? The approach will be based on that.

But if you just want to build transaction and post, you can do so by following options

1. using Ogmios and Kupo connecting to local node. There is a backend service for this. But you need to run these two processes.

2. Implementing your own UtxoSupplier and ProtocolParamSupplier interface.  

So the tricky part here is how to connect to a Cardano node from Java. You can use Yaci to do that as it's a Cardano mini protocol implementation in java. 

https://github.com/bloxbean/yaci-core

Recently, I had written a sample to build / post transaction to a local cardano node using Yaci. Basically another impl for UtxoSupplier and ProtocolParams. Fyi, it's using a slightly older version of Yaci.

https://github.com/bloxbean/ccl-local-tx-example

This was just a poc. Few things are still pending.  Like parsing of transaction error message is not done yet. So you just get a cbor string.

But it's in the todo list, hopefully once the  tx submission error decoding is done properly in yaci, we can add it to cardano-client.
